src/auth/oauth2/gui/qgsauthoauth2edit.ui

```diff
@@ -637,6 +637,146 @@
+               <string>Extra token(s) header</string>
+              </property>
+              <property name="wordWrap">
+               <bool>true</bool>
+              </property>
+             </widget>
+            </item>
+            <item row="19" column="1">
+             <layout class="QHBoxLayout" name="horizontalLayout_6">
+              <property name="spacing">
+               <number>3</number>
+              </property>
+              <item>
+               <widget class="QTableWidget" name="mExtraTokensTable">
+                <property name="sizePolicy">
+                 <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
+                  <horstretch>1</horstretch>
+                  <verstretch>0</verstretch>
+                 </sizepolicy>
+                </property>
+                <property name="minimumSize">
+                 <size>
+                  <width>50</width>
+                  <height>70</height>
+                 </size>
+                </property>
+                <property name="editTriggers">
+                 <set>QAbstractItemView::AllEditTriggers</set>
+                </property>
+                <property name="dragEnabled">
+                 <bool>true</bool>
+                </property>
+                <property name="dragDropMode">
+                 <enum>QAbstractItemView::DragOnly</enum>
+                </property>
+                <property name="selectionMode">
+                 <enum>QAbstractItemView::SingleSelection</enum>
+                </property>
+                <property name="sortingEnabled">
+                 <bool>true</bool>
+                </property>
+                <property name="wordWrap">
+                 <bool>false</bool>
+                </property>
+                <attribute name="horizontalHeaderMinimumSectionSize">
+                 <number>120</number>
+                </attribute>
+                <attribute name="horizontalHeaderDefaultSectionSize">
+                 <number>120</number>
+                </attribute>
+                <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
+                 <bool>true</bool>
+                </attribute>
+                <attribute name="horizontalHeaderStretchLastSection">
+                 <bool>true</bool>
+                </attribute>
+                <attribute name="verticalHeaderVisible">
+                 <bool>false</bool>
+                </attribute>
+                <column>
+                 <property name="text">
+                  <string>Token</string>
+                 </property>
+                </column>
+                <column>
+                 <property name="text">
+                  <string>Header name</string>
+                 </property>
+                </column>
+                <property name="toolTip">
+                 <string>These extra tokens will be included in the request headers sent to the resource when provided by the token endpoint</string>
```

Comment:
@nyalldawson the first column refers to extra tokens provided by the authorization service. Can we do better with this wording or the UI to make this clear?

Reply:
@nyalldawson , the first column is not free-form string, it is delivered by the token endpoint as part of its json response. E.g., you can see how Google defines the token/tag key names in its documentation: https://developers.google.com/identity/openid-connect/openid-connect#exchangecode

Reply:
ok -- i think there's a naming issue causing the confusion. To me "token" sounds like it is a constant value, which will be inserted as is into the request.

How do you feel about flipping the columns and renaming to:

Request Header Name | Token Response Field

?

Reply:
@nyalldawson , sounds reasonable to me. It'd do a better job at representing the outcome by placing the header first.